### DISCUSSION TITLE
Getting Notifications of Expiring Certs

### DISCUSSION THREAD

**Discussion by lunakoa** [chris2511/xca#566] [Ideas]
We tie XCA into our nagios system, in short we do a query and export all the certs and we get notified when they expire.

I would like to modify the database by adding a custom nagios table to generate custom settings for certain fields needed in nagios, like who to notify.

It works really well for us, verisgn, lets encrypt, internal CA's.  Kind of cool.  Was thinking of doing a video on it.

**Comment by chris2511** [comment]
There is the possibility to export a complete CA as ICS Calendar entries containing all expiration dates of all issued certificates and the most recent CRL, if that helps.

Adding an additional table shold not harm, not even during a schema update when opening a database with a newer version of XCA. However, it will be completely "invisible" to XCA. And obviously even deleting items will not delete entries from this table, so be careful with "foreign keys".

Alternatively, you may use a line in the comment, prefixed with a tag like "NAGIOS: notifyuser=...".
This solution is also a little ugly, but differently ugly :-)

**Comment by LongBlackRoad** [comment]
What would be nice is to have link to an SMTP server. Then you can receive email alerts when certificates are going to expire.
Thoughts?
